packages/spectacular-docs/docs/pipe-testing/index.md

```diff
@@ -0,0 +1,254 @@
+---
+id: index
+slug: /pipe-testing
+sidebar_label: Introduction
+title: Pipe testing
+---
+
+Spectacular's pipe testing API configures the Angular testing module and sets up
+a host component for the Angular pipe under test.
+
+The pipe test harness makes it straightforward to test pipes in the same way as
+they are used in a component template. This gives us the confidence that our
+Angular pipe works as intended.
+
+## Example Angular pipe
+
+As an example, try imagining how to test the Angular pipe in the following code
+snippet:
+
+```ts
+// pow.pipe.ts
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({ name: 'pow' })
+export class PowPipe implements PipeTransform {
+  transform(base: number, exponent = 1): number {
+    return Math.pow(base, exponent);
+  }
+}
+```
+
+## Testing an Angular pipe in isolation with the Angular testbed
+
+We could unit test it in isolation by exercising its `transform` method as seen
+in the following example:
+
+```ts {18}
+import { TestBed } from '@angular/core/testing';
+
+import { PowPipe } from './pow.pipe';
+
+describe(PowPipe.name, () => {
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [PowPipe],
+    });
+    pipe = TestBed.inject(PowPipe);
+  });
+
+  let pipe: PowPipe;
+
+  it('raises the base to the power of 1 by default', () => {
+    const base = 2;
+
+    const actual = pipe.transform(base);
+
+    expect(actual).toBe(2);
+  });
+});
+```
+
+How would we pass additional options? This is easy in an isolated unit test as
+seen in the following test case:
+
+```ts {5}
```

Comment:
This is new to me, what do the numbers inside the `{}`

Reply:
They highlight one or more lines of code, in this case the 5th line in this code block.